pkg/validate/openshiftcluster_validatedynamic.go

```diff
@@ -263,6 +258,11 @@ func (dv *openShiftClusterDynamicValidator) Dynamic(ctx context.Context) error {
 		return err
 	}
 
+	err = fpDynamic.ValidatePreConfiguredNSGs(ctx, dv.oc, subnets)
```

Comment:
The goal was to group the specific types of validation together.  We should try to keep the service-principal validations together as much as possible.  I would rather move the SP validations to the top than move a singular FP validation to the bottom.

Reply:
Agreed.  I added more description on how the error currently occurs (TLDR: any calls to `ValidatePreconfiguredNSG` needs to happen after `ValidateSubnets`.  This should no longer be the case after GA as the logic in `ValidateSubnets` will go away).  

I moved all the SP validations to the top and FP's lower to keep them grouped.